tests/db2/CMakeLists.txt

```diff
@@ -1,16 +1,9 @@
-###############################################################################
-#
-# This file is part of CMake configuration for SOCI library
-#
-# Copyright (C) 2010-2013 Mateusz Loskot <mateusz@loskot.net>
-# Distributed under the Boost Software License, Version 1.0.
-# (See accompanying file LICENSE_1_0.txt or copy at
-# http://www.boost.org/LICENSE_1_0.txt)
-#
-###############################################################################
+add_executable(soci_db2_tests OBJECT db2_tests.cpp)
+target_link_libraries(soci_db2_tests PRIVATE soci_common_tests soci_db2_interface)
 
-soci_backend_test(
```

Comment:
Those functions were more or less the prime reason why the SOCI cmake files were so unreadable. Every time you encounter such a function, you need to look up what exactly it does (and these functions do a lot of implicit stuff that becomes important when one is interested in the details).

And most often when looking into cmake files, one does not have a language server that can do a quick "show definition". Instead, one has to manually search where this function might be defined in order to figure out what it does.

For this reason I chose to mostly remove functions such as this one from the cmake setup. Where I do use functions, I try to have them limited in scope such that their name hopefully is enough to know all about what the function does so that a lookup is mostly unnecessary.